unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,439 @@
+                 return_value=TEST_JWT_KEY_FROM_SERVER)
+    mocker.patch("gsy_framework.client_connections.utils"
+                 ".RepeatingTimer")
+
+
+@pytest.fixture(name="mock_grid_fee_calculation")
+def fixture_mock_grid_fee_calculation(mocker):
+    """Mock GridFeeCalculation class."""
+    mocker.patch("gsy_e_sdk.aggregator.GridFeeCalculation")
+
+
+@pytest.fixture(name="mock_environment_use_functions")
+def fixture_mock_env_use_functions(mocker):
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.simulation_id_from_env",
+                 return_value=TEST_SIMULATION["uuid"])
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.domain_name_from_env",
+                 return_value=TEST_SIMULATION["domain_name"])
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.websocket_domain_name_from_env",
+                 return_value=TEST_SIMULATION["websockets_domain_name"])
+
+
+@pytest.fixture(name="mock_outgoing_funcs_in_construction")
+def fixture_mock_outgoing_funcs_in_construction(mocker):
+    test_aggregators_list = [{"name": TEST_AGGREGATOR_NAME, "uuid": TEST_AGGREGATOR_UUID},
+                             {"name": TEST_AGGREGATOR_NAME + "2", "uuid": None}]
+    test_new_aggregator_dict = [{"name": TEST_AGGREGATOR_NAME, "uuid": TEST_AGGREGATOR_UUID}]
+
+    mocker.patch("gsy_e_sdk.aggregator.blocking_get_request",
+                 return_value=test_aggregators_list)
+    mocker.patch("gsy_e_sdk.aggregator.blocking_post_request",
+                 return_value=test_new_aggregator_dict)
+    mocker.patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver")
+    mocker.patch("gsy_e_sdk.aggregator.WebsocketThread")
+    mocker.patch("gsy_e_sdk.aggregator.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_execute_batch_command_methods")
+def fixture_mock_execute_batch_command_methods(mocker):
+    """Mock methods and external functions and other methods used in
+    execute_batch_command."""
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
+                 new_callable=PropertyMock, return_value=3)
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.execute_batch",
+                 return_value=TEST_BATCH_COMMAND_DICT)
+    mocker.patch("gsy_framework.client_connections.utils.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+    mocker.patch("gsy_framework.client_connections.utils.post_request",
+                 return_value=True)
+
+
+@pytest.fixture(name="aggregator")
+def fixture_aggregator(mock_outgoing_funcs_in_construction):  # pylint: disable=unused-argument
+    return Aggregator(aggregator_name=TEST_AGGREGATOR_NAME,
+                      simulation_id=TEST_SIMULATION["uuid"],
+                      domain_name=TEST_SIMULATION["domain_name"],
+                      websockets_domain_name=TEST_SIMULATION["websockets_domain_name"]
+                      )
+
+
+class TestAggregatorConstructor:
```

Comment:
Can you please remind me how the outcome of the discussion about the place of the test functions was? What I mean is, whether we agreed on putting all tests methods into one class or in several or whether we just stick to put these into one module for each tested module/class.

Reply:
(Note for Eduardo: after this comment we've discussed this on slack. I've also updated the best practices page on Confluence to reflect the conversation outcomes).

Reply:
As done [here](https://github.com/gridsingularity/gsy-e-sdk/pull/197/commits/a180cac52b4ffeabb4363dcab6ec22882249357d), I'll commit a change in which I group all test methods in a single class in order to avoid having too many classes with only one test method.